src/app/trip-page/trip-photos-list/trip-photos-list.component.ts

```diff
@@ -0,0 +1,142 @@
+  displayedFullPhoto$!: Observable<TripPhoto>;
+  displayedFullPhotoIndex$!: Observable<number>;
+
+  constructor(
+    private tripPhotoService: TripPhotoService,
+    private changeDetection: ChangeDetectorRef
+  ) {}
+
+  detect() {
+    this.changeDetection.detectChanges();
+  }
+
+  select() {
+    let selectedCount = 0;
+    this.photos.forEach((photoElement: ElementRef) => {
+      if (photoElement.nativeElement.checked) {
+        selectedCount++;
+      }
+    })
+    if (selectedCount === 0) {
+      this.toggleSelectAll$.next(false);
+      this.disableDeleteButton$.next(true);
+    } else {
+      this.toggleSelectAll$.next(true);
+      this.disableDeleteButton$.next(false);
+    }
+    this.selectedCount$.next(selectedCount);
+    this.detect(); // updates UI for use case that photo(s) were deleted or uploaded before selecting
+  }
+
+  displayFullViewImg(photo: TripPhoto, index: number): void {
+    this.displayedFullPhotoIndex$ = of(index);
+    this.displayFullView$ = of(true);
+    this.displayedFullPhoto$ = of(photo);
+    this.detect(); // updates UI for use case that photo(s) were deleted or uploaded before selecting
+  }
+
+  selectAll(): void {
+    // if not all selected already --> select all
+      if (this.toggleSelectAll$.getValue() === false) {
+        for (let p of this.photos) {
+          p.nativeElement.checked = true;
+        }
+        this.selectedCount$.next(this.photos.length);
+        this.toggleSelectAll$.next(true);
+        this.disableDeleteButton$.next(false);
+      } else {
+        // clear selection
+        for (let p of this.photos) {
+          p.nativeElement.checked = false;
+        }
+        this.selectedCount$.next(0);
+        this.toggleSelectAll$.next(false);
+        this.disableDeleteButton$.next(true);
+      }
+      this.detect(); // updates UI for use case that photo(s) were deleted or uploaded before selecting
+  }
+
+  removePhotos(): void {
+    let photoFileNames: string[] = [];
+    let photosToRemove: TripPhoto[] = [];
+
+    // create array with filenames of selected photos
+    for (let p of this.photos) {
+      if (p.nativeElement.checked) {
+        photoFileNames.push(p.nativeElement.id);
+      }
+    }
+
+    // validation
+    if (photoFileNames.length < 1) {
+      return;
+    }
+    // update UI that photo removal process has began
+    this.progressMessage$.next('Removing photos...');
+
+    // grab photo files by filenames in filename array
+    this.tripPhotos$.pipe(take(1)).subscribe((photos) => {
+      for (let pFileName of photoFileNames) {
+        let tripPhotoObj = photos.find((tripPhoto) => tripPhoto.fileName === pFileName)!;
+        if (tripPhotoObj !== undefined) {
+          photosToRemove.push(tripPhotoObj);
+        }
+      }
+
+      // delete photos via API
+      this.tripPhotoService
+        .deletePhotos(photosToRemove, photos[0].tripId)
+        .pipe(take(1))
+        .subscribe({
+          next: (updatedTrip) => {
```

Comment:
`next` and `error` are a style that we really shouldn't be using. use `catchError` for errors and just use the body of the subscribe for `next`